clippy_lints/src/operators/arithmetic_side_effects.rs

```diff
@@ -184,6 +184,35 @@ impl ArithmeticSideEffects {
         }
     }
 
+    /// There are some integer methods like `wrapping_div` that will panic depending on the
+    /// provided input.
+    fn manage_method_call<'tcx>(
+        &mut self,
+        args: &[hir::Expr<'tcx>],
+        cx: &LateContext<'tcx>,
+        instance: &hir::Expr<'tcx>,
+        ps: &hir::PathSegment<'tcx>,
+    ) {
+        const METHODS: &[&str] = &["saturating_div", "wrapping_div", "wrapping_rem", "wrapping_rem_euclid"];
+        let Some(arg) = args.first() else { return; };
+        if constant_simple(cx, cx.typeck_results(), instance).is_some() {
+            return;
+        }
+        let instance_ty = cx.typeck_results().expr_ty(instance);
+        if !Self::is_integral(instance_ty) {
+            return;
+        }
+        let has_method = METHODS.iter().copied().any(|method| method == ps.ident.as_str());
+        if !has_method {
+            return;
+        }
+        let (actual_arg, _) = peel_hir_expr_refs(arg);
```

Comment:
The `arg` is used here, so it would be better that is's moved here.
```suggestion
        let Some(arg) = args.first() else { return; };
        let (actual_arg, _) = peel_hir_expr_refs(arg);
```